step-templates/azure-function-set-appsettings-from-azconfig.json

```diff
@@ -0,0 +1,149 @@
\"\n    Write-Verbose \"Resource Group: $ResourceGroup\"\n    Write-Verbose \"Slot: $Slot\"\n    Write-Verbose \"Settings: $($Settings.Count)\"\n    Write-Verbose \"Additional Settings: $($AdditionalSettings.Count)\"\n    if ($AdditionalSettings.Count -gt 0) {\n        Write-Verbose \"Combining additional settings with settings retrieved from Azure App Config\"\n        $Settings = $Settings + $AdditionalSettings\n    }\n\n    $settingsFile = $null\n\n    try {\n\n        $command = \"az functionapp config appsettings set --name=\"\"$Functionname\"\" --resource-group \"\"$ResourceGroup\"\" \"\n        if (-not([string]::IsNullOrWhiteSpace($Slot))) {\n            $command += \" --slot \"\"$Slot\"\" \"\n        }\n\n        if ($Settings.Count -ge 1) {\n            $settingsFile = [System.IO.Path]::GetRandomFileName()\n            $ConvertToJsonParameters = @{}\n            if ($PSVersionTable.PSVersion.Major -ge 6) {\n                $ConvertToJsonParameters.Add(\"AsArray\", $True)\n            }\n            $Settings | ConvertTo-Json @ConvertToJsonParameters | Set-Content -Path $settingsFile\n            $command += \" --settings '@$settingsFile'\"\n        }\n\n        Write-Verbose \"Invoking expression: $command\"\n        Write-Host \"##octopus[stderr-progress]\"\n        $settingsUpdateResponse = Invoke-Expression -Command $command\n        $ExitCode = $LastExitCode\n        Write-Verbose \"FunctionApp update ExitCode: $ExitCode\"\n        if ($ExitCode -ne 0) {\n            throw \"Error configuring appsettings for function app '$($FunctionName)'. ExitCode: $ExitCode\"\n        }\n        Write-Host \"##octopus[stderr-default]\"\n        if ($null -ne $settingsUpdateResponse) {\n            Write-Host \"Update of function '$($FunctionName)' was successful\"\n            try {\n                $functionSettings = $settingsUpdateResponse | ConvertFrom-Json\n                if ($null -ne $functionSettings) {\n                    $settingsCount = @($functionSettings | Where-Object { $_.slotSetting -eq $False }).Count\n                    $slotSettingsCount = @($functionSettings | Where-Object { $_.slotSetting -eq $True }).Count\n                    Write-Verbose \"Function '$($FunctionName)' has $($settingsCount) setting(s) and $($slotSettingsCount) slot setting(s).\"\n                }\n            }\n            catch {}\n        }\n        \n    }\n    catch { throw }\n    finally {\n        if (-not([string]::IsNullOrWhiteSpace($settingsFile))) {\n            Write-Verbose \"Removing temporary settings file $settingsFile\"\n            Remove-Item -Path $settingsFile -Force -ErrorAction Ignore\n        }\n        if (-not([string]::IsNullOrWhiteSpace($slotSettingsFile))) {\n            Write-Verbose \"Removing temporary slot settings file $slotSettingsFile\"\n            Remove-Item -Path $slotSettingsFile -Force -ErrorAction Ignore\n        }\n    }\n}\nelse {\n    Write-Host \"No settings found to publish to the Azure App function\"\n}",
```

Comment:
This line:

```
$settingName = "$($keyName.Trim())"
```

can be simplified to:

```
$settingName = $keyName.Trim()
```

Reply:
`if ($null -eq $appConfigResponse) {` should be changed to `if ([string]::IsNullOrWhiteSpace($appConfigResponse)) {` to catch empty responses.

Reply:
There are a few boolean checks like `if ($RetrieveAllKeys -eq $True)` that could be simplified.

Reply:
The additional brackets around variables like `"Unable to establish additional setting name from: '$($_)'"` are redundant in a few places:

* `"Unable to establish additional setting name from: '$($_)'"`
* `"Function '$($FunctionName)' has $($settingsCount) setting(s) and $($slotSettingsCount) slot setting(s)."`
* `"Error configuring appsettings for function app '$($FunctionName)'. ExitCode: $ExitCode"`